scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1466,143 +1470,174 @@
+            ; by default, do not set disk-limit
+            (when disk-limit
+              (.putLimitsItem resources "ephemeral-storage" (double->quantity (* constraints/disk-multiplier disk-limit)))))
+          (add-node-selector pod-spec (pool->disk-type-label-name pool-name) disk-type))
         (.setResources container resources)
-        (.addInitContainersItem pod-spec container)))
+        (.setVolumeMounts container (filterv some? (conj (concat volume-mounts main-container-checkpoint-volume-mounts)
+                                                         (init-container-workdir-volume-mount-fn true)
+                                                         (scratch-space-volume-mount-fn false)
+                                                         (sidecar-workdir-volume-mount-fn true))))
+        (.setWorkingDir container workdir)
+
+        (.addContainersItem pod-spec container))
 
-    ; sandbox file server container
-    (when use-cook-sidecar?
-      (let [{:keys [command health-check-endpoint image image-fallback port resource-requirements]} sidecar
-            {:keys [cpu-request cpu-limit memory-request memory-limit]} resource-requirements
-            container (V1Container.)
-            resources (V1ResourceRequirements.)]
-        ; container
-        (.setName container cook-container-name-for-file-server)
-        (.setImage container (resolve-image-from-incremental-config
-                               job passport-event-base passport/sidecar-image-selected
-                               image image-fallback))
-        (.setCommand container (conj command (str port)))
-        (.setWorkingDir container sidecar-workdir)
-        (.setPorts container [(.containerPort (V1ContainerPort.) (int port))])
-
-        (.setEnv container (conj main-env-vars
-                                 ;; DEPRECATED - sidecar should use COOK_SANDBOX instead.
-                                 ;; Will remove this environment variable in a future release.
-                                 (make-env "COOK_WORKDIR" sandbox-dir)))
-
-        ; optionally enable http-based readiness probe
-        (when health-check-endpoint
-          (let [http-get-action (doto (V1HTTPGetAction.)
-                                  (.setPort (-> port int IntOrString.))
-                                  (.setPath health-check-endpoint))
-                readiness-probe (doto (V1Probe.)
-                                  (.setHttpGet http-get-action))]
-            (.setReadinessProbe container readiness-probe)))
-
-        ; resources
-        (set-mem-cpu-resources resources memory-request memory-limit cpu-request cpu-limit)
-        (.setResources container resources)
 
-        (.setVolumeMounts container [(sandbox-volume-mount-fn true) (sidecar-workdir-volume-mount-fn false) (scratch-space-volume-mount-fn false)])
-        (.addContainersItem pod-spec container)))
+      (let [get-resource-requirements-fn (fn [fieldname] (if use-cook-sidecar?
```

Comment:
Groktest: you're refactoring to have some common variables between the init-container and the new checkpoint-init-container

Reply:
Yup. I could have copy&pasted the whole block, but that would have had a lot of duplicate variable definitions.